### ISSUE TITLE
Implement a more Advanced model

### ISSUE THREAD

**Issue by ViggoJahr** [ViggoJahr/soccer-prediction-project#13]
Add a different approach (e.g., Random Forest, XGBoost) alongside the baseline model

Compare performance metrics vs. the baseline (e.g., confusion matrix, accuracy, recall)

**Comment by amjgp** [comment]
Implemented a model with hyperparameter tuning with the XGBoost library
